rekuho/src/lib/Portion.svelte

```diff
@@ -59,20 +61,58 @@
 
   const transformToScreenPosition = ([x, y]: Vector2): Vector2 => [x * columnWidth, y * lineHeight]
 
+  ////////////////////////////////////////////////////////////////////////////////
+
   const longestLine = (text: string[]): string =>
     text.reduce((a, b) => (a.length < b.length ? b : a), "")
 
   $: linesViewWidth = $state.lines ? longestLine($state.lines).length : 1
   $: textViewWidth = width - gutterWidth
-  $: textToVisibleRatio = (linesViewWidth * columnWidth - theme.textOffset) / width
-  $: verticalScrollerWidth = textViewWidth / textToVisibleRatio
+  $: textWidthToVisibleRatio = (linesViewWidth * columnWidth - config.textOffset) / width
+  $: horizontalScrollerWidth = textViewWidth / textWidthToVisibleRatio
+
+  ////////////////////////////////////////////////////////////////////////////////
+
+  let scrollOffset: pixels = 0
+
+  // const scrollShowLine = (lineNumber: line, height: pixels): void => {
+  //   const lineOffset = lineNumber * lineHeight
+  //   if (lineOffset + lineHeight > scrollOffset + height) {
+  //     scrollOffset = lineOffset - height + lineHeight
+  //   } else if (scrollOffset > lineOffset) {
+  //     scrollOffset = lineOffset
+  //   }
+  // }
+
+  // $: scrollShowLine($state.carets[0]?.line ?? 0, height)
```

Comment:
why is there weird commented out code here? scrolling a line into view should be a backend responsibility, thus this shouldn't really ever be required in the frontend